cmd/source_profile.go

```diff
@@ -0,0 +1,274 @@
+	}
+	if pwd, ok := params["password"]; ok {
+		mysql.pwd = pwd
+	}
+	return mysql
+}
+
+type SourceProfileConnectionPostgreSQL struct {
+	host string // Same as PGHOST environment variable
+	port string // Same as PGPORT environment variable
+	user string // Same as PGUSER environment variable
+	db   string // Same as PGDATABASE environment variable
+	pwd  string // Same as PGPASSWORD environment variable
+}
+
+func NewSourceProfileConnectionPostgreSQL(params map[string]string) SourceProfileConnectionPostgreSQL {
+	pg := SourceProfileConnectionPostgreSQL{}
+	if host, ok := params["host"]; ok {
+		pg.host = host
+	}
+	if port, ok := params["port"]; ok {
+		pg.port = port
+	}
+	if user, ok := params["user"]; ok {
+		pg.user = user
+	}
+	if db, ok := params["db_name"]; ok {
+		pg.db = db
+	}
+	if pwd, ok := params["password"]; ok {
+		pg.pwd = pwd
+	}
+	return pg
+}
+
+type SourceProfileConnectionDynamoDB struct {
+	awsAccessKeyID     string // Same as AWS_ACCESS_KEY_ID environment variable
+	awsSecretAccessKey string // Same as AWS_SECRET_ACCESS_KEY environment variable
+	awsRegion          string // Same as AWS_REGION environment variable
+	dydbEndpoint       string // Same as DYNAMODB_ENDPOINT_OVERRIDE environment variable
+	schemaSampleSize   int64  // Number of rows to use for inferring schema (default 100,000)
+}
+
+func NewSourceProfileConnectionDynamoDB(params map[string]string) (SourceProfileConnectionDynamoDB, error) {
+	// TODO: Populate environment variables for DynamoDB if passed as flags.
+	dydb := SourceProfileConnectionDynamoDB{}
+	if awsAccessKeyID, ok := params["awsAccessKeyID"]; ok {
+		dydb.awsAccessKeyID = awsAccessKeyID
+	}
+	if awsSecretAccessKey, ok := params["awsSecretAccessKey"]; ok {
+		dydb.awsSecretAccessKey = awsSecretAccessKey
+	}
+	if awsRegion, ok := params["awsRegion"]; ok {
+		dydb.awsRegion = awsRegion
+	}
+	if dydbEndpoint, ok := params["dydbEndpoint"]; ok {
+		dydb.dydbEndpoint = dydbEndpoint
+	}
+	if schemaSampleSize, ok := params["schemaSampleSize"]; ok {
+		schemaSampleSizeInt, err := strconv.Atoi(schemaSampleSize)
+		if err != nil {
+			return dydb, fmt.Errorf("could not parse schemaSampleSize = %v as a valid int64", schemaSampleSize)
+		}
+		dydb.schemaSampleSize = int64(schemaSampleSizeInt)
+	}
+	return dydb, nil
+}
+
+type SourceProfileConnection struct {
+	ty    SourceProfileConnectionType
+	mysql SourceProfileConnectionMySQL
+	pg    SourceProfileConnectionPostgreSQL
+	dydb  SourceProfileConnectionDynamoDB
+}
+
+func NewSourceProfileConnection(source string, params map[string]string) (SourceProfileConnection, error) {
+	conn := SourceProfileConnection{}
+	switch strings.ToLower(source) {
+	case "mysql":
+		{
+			conn.ty = SourceProfileConnectionTypeMySQL
+			conn.mysql = NewSourceProfileConnectionMySQL(params)
+		}
+	case "postgresql", "postgres", "pg":
+		{
+			conn.ty = SourceProfileConnectionTypePostgreSQL
+			conn.pg = NewSourceProfileConnectionPostgreSQL(params)
+		}
+	case "dynamodb":
```

Comment:
Should we add dydb too?

Reply:
Not sure if dydb is common enough to warrant adding support similar to postgres or pg. Does it make sense to defer this and add the same when users demand for the same?

Reply:
I will leave it as it is for now, but should be a quick follow up change if required.